engine/lite/kafka/runtime/src/main/scala/pl/touk/nussknacker/engine/lite/kafka/NuKafkaRuntimeApp.scala

```diff
@@ -45,25 +51,35 @@ object NuKafkaRuntimeApp extends App with LazyLogging {
   }
   System.exit(exitCode)
 
+  private val akkaHttpCloseTimeout = 10 seconds
+
   private def runAfterActorSystemCreation(): Unit = {
     val scenarioInterpreter = RunnableScenarioInterpreterFactory.prepareScenarioInterpreter(scenario, runtimeConfig, liteKafkaJobData, system)
     Runtime.getRuntime.addShutdownHook(new Thread() {
       override def run(): Unit = {
+        logger.info("Closing http server")
+        if (server != null) Await.result(server.terminate(akkaHttpCloseTimeout), akkaHttpCloseTimeout)
         logger.info("Closing RunnableScenarioInterpreter")
         scenarioInterpreter.close()
       }
     })
 
-    val healthCheckServer = new HealthCheckServerRunner(system, scenarioInterpreter)
-    Await.result(for {
-      _ <- healthCheckServer.start()
-      _ <- scenarioInterpreter.run()
-    } yield (), Duration.Inf)
+    val healthCheckProvider = new HealthCheckRoutesProvider(system, scenarioInterpreter)
+
+    val httpServer = Http().newServerAt(interface = httpConfig.interface, port = httpConfig.port)
 
+    val run = scenarioInterpreter.run()
+    val healthCheckRoutes = healthCheckProvider.routes()
+    val routes = scenarioInterpreter.routes().map(_ ~ healthCheckRoutes).getOrElse(healthCheckRoutes)
+    httpServer.bind(routes).foreach { b =>
```

Comment:
Please sequence `run` with `bind` - otherwise we won't see binding failures in await